pkg/transportrequest/rfc/upload.go

```diff
@@ -0,0 +1,152 @@
+package rfc
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/config/validation"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+const (
+	eq = "="
+)
+
+// Exec Everything we need for calling an executable
+type Exec interface {
+	command.ExecRunner
+	GetExitCode() int
+}
+
+// Upload ...
+type Upload interface {
+	Perform(Exec) error
+	WithConnection(Connection)
+	WithConfiguration(UploadConfig)
+	WithApplication(Application)
+	WithTransportRequestID(string)
+	WithApplicationURL(string)
+}
+
+// Connection Everything we need for connecting to the ABAP system
+type Connection struct {
+	// The endpoint in for form <protocol>://<host>:<port>, no path
+	Endpoint string
+	// The ABAP client, like e.g. "001"
+	Client string
+	// The ABAP instance, like e.g. "DEV",  "QA"
+	Instance string
+	User     string
+	Password string
+}
+
+// Application Application specific properties
+type Application struct {
+	Name        string
+	Description string
+	AbapPackage string
+}
+
+// UploadConfig additional configuration properties
+type UploadConfig struct {
+	AcceptUnixStyleEndOfLine bool
+	CodePage                 string
+	FailUploadOnWarning      bool
+	Verbose                  bool
+}
+
+// UploadAction Collects all the properties we need for the deployment
+type UploadAction struct {
+	Connection         Connection
+	Application        Application
+	Configuration      UploadConfig
+	TransportRequestID string
+	ApplicationURL     string
+}
+
+// WithApplicationURL The location of the deployable
+func (action *UploadAction) WithApplicationURL(z string) {
+	action.ApplicationURL = z
+}
+
+// WithTransportRequestID The transport request ID for the upload
+func (action *UploadAction) WithTransportRequestID(t string) {
+	action.TransportRequestID = t
+}
+
+// WithApplication Everything we need to know about the application
+func (action *UploadAction) WithApplication(a Application) {
+	action.Application = a
+}
+
+// WithConfiguration Everything we need to know in order to perform the upload
+func (action *UploadAction) WithConfiguration(c UploadConfig) {
+	action.Configuration = c
+}
+
+// WithConnection Everything we need to know about the connection
+func (action *UploadAction) WithConnection(c Connection) {
+	action.Connection = c
+}
+
+// Perform Performs the upload
+func (action *UploadAction) Perform(command Exec) error {
+
+	log.Entry().Infof("Deploying artifact '%s' to '%s', client: '%s', instance: '%s'.",
+		action.ApplicationURL, action.Connection.Endpoint, action.Connection.Client, action.Connection.Instance,
+	)
+
+	missingParameters, err := validation.FindEmptyStringsInConfigStruct(*action)
+	if err != nil {
+		return fmt.Errorf("Cannot check that all required parameters are available for SOLMAN upload: %w", err)
```

Comment:
how can this happen? why "cannot check"? looks like 
return `fmt.Errorf("invalid configuration parameters detected: %w", err)`

Reply:
Pipeline owners can do nothing there. The error I would expect is: wrong type found during traversal which [cannot be handled](https://github.com/SAP/jenkins-library/blob/06afce300500c17058ef6993a1cf3d3de80743a4/pkg/config/validation/validation.go#L41). I would expect such situations are found during the development process during testing. The original exception is wrapped by the exception which is created here ("%w"), so the details should be provided. Don't know how we can come to a more helpful message. Suggestions welcome ...

Reply:
Proposal: "invalid configuration parameters detected. SOLMAN upload parameter may be missing : %w",

Reply:
done